code/frameworks/nextjs/src/images/next-image-default-loader.tsx

```diff
@@ -24,5 +24,17 @@ export const defaultLoader = ({ src, width, quality }: _NextImage.ImageLoaderPro
     );
   }
 
-  return `${src}?w=${width}&q=${quality ?? 75}`;
+  const [baseUrlAndSearch, hash = ''] = src.split('#');
+  const [baseUrl, search = ''] = baseUrlAndSearch.split('?');
+
```

Comment:
Hey, I helped @json-betsec with this PR. Thanks for the feedback!

We avoided the `URL` API because `src` can be relative, and the `URL` API can't handle relative URLs without a base URL. We weren't sure if it was always possible to use something like `location.href` as the base URL, for example, if the component needs to be server-side renderable.

Even if we always have a usable base URL, it would be pretty tricky to extract the original "relative" path from the final URL - for example in the case of resolving a `src` _without_ a leading slash like `foo/bar.png` against `https://example.org/some/path/without-a-trailing-slash`, if we wanted the final URL to remain relative (i.e. not starting with a leading slash), then the implementation would become non-trivial. _Maybe_ that's not a big deal, but we felt like `src` should stay relative in order to minimise the chance it changes the behaviour of `next/image`.

We also thought about suggesting a much simpler approach - why not just... **not** add these query params at all? We assumed there must have been some intention there to support an image resizing system of some kind. But `src` can be _literally_ any URL. There's no way to guarantee that adding arbitrary query params won't cause issues with some external image server (e.g. `q` might be interpreted as a search string).

Anyway, if adding the query params is still desirable, I have a reasonable amount of confidence that the PR code here will work for all URLs (once upon a time I wrote a URI parsing library). But failing that, we could perhaps use an NPM package with support for relative URLs like `url-parse`?

Reply:
I found a way to get away with using `URL` preserving relative paths! Let me know any thoughts about these parameters though

Reply:
@ndelangen @yannbf Any ideas if this code also needs to work on the server?

Reply:
This is client-side only:
https://github.com/storybookjs/storybook/blob/4d728c3b46867b0445a30a578706bedc7e54115a/code/frameworks/nextjs/src/images/next-image.tsx#L19
https://github.com/storybookjs/storybook/blob/4d728c3b46867b0445a30a578706bedc7e54115a/code/frameworks/nextjs/src/images/next-legacy-image.tsx#L20